rocminfo.cc

```diff
@@ -459,18 +459,17 @@ static void DisplayAgentInfo(agent_info_t *agent_i) {
     for (int i = 0; i < 3; i++) {
       dim = "Dim[" + std::to_string(i) + "]:";
       printLabelInt(dim.c_str(),
-              reinterpret_cast<uint32_t*>(&agent_i->workgroup_max_dim)[i], 2);
+              static_cast<uint32_t>(agent_i->workgroup_max_dim[i]), 2);
     }
     printLabelInt("Grid Max Size:", agent_i->grid_max_size, 1);
     printLabelInt("Waves Per CU:", agent_i->waves_per_cu, 1);
     printLabelInt("Max Work-item Per CU:",
                             agent_i->wavefront_size*agent_i->waves_per_cu, 1);
+
     printLabel("Grid Max Size per Dimension:", true, 1);
-    for (int i = 0; i < 3; i++) {
-      dim = "Dim[" + std::to_string(i) + "]:";
-      printLabelInt(dim.c_str(),
-                 reinterpret_cast<uint32_t*>(&agent_i->grid_max_dim)[i], 2);
-    }
+    printLabelInt("Dim[0]", agent_i->grid_max_dim.x, 2);
+    printLabelInt("Dim[1]", agent_i->grid_max_dim.y, 2);
+    printLabelInt("Dim[2]", agent_i->grid_max_dim.z, 2);
```

Comment:
If you are going to unroll the loop, you may as well use the HSA naming ("x, y z") instead of "Dim[0], Dim[1], Dim[2]"

Reply:
For the workgroup max sizes for both the agent the isa info, it's using the "Dim[n]" naming.  I don't have any strong preference between "x,y,z" and "Dim[n]" but I think they should be consistent.

Reply:
I agree, consistency is most important. Since the documentation refers to (x, y, z), then I would go with that (which means not using a loop in the workgroup cases too)...assuming this doesn't cause any complications. If the "Dim[]" is somehow easier than "x, y, z" on tools that parse this output, then leaving it as-is is fine.